instantsearch/src/commonMain/kotlin/com/algolia/instantsearch/searcher/hits/internal/DefaultHitsSearcher.kt

```diff
@@ -59,7 +65,14 @@ internal class DefaultHitsSearcher(
     override fun searchAsync(): Job {
         return coroutineScope.launch(exceptionHandler) {
             isLoading.runAsLoading {
-                response.value = search()
+                val responseSearch = search()
+                response.value = responseSearch
+                if (responseSearch != null && isAutoSendingHitsViewEvents) {
+                    val events = getViewEvents(responseSearch)
+                    if (events.isNotEmpty()) {
+                        insights.sendEvents(events)
+                    }
+                }
```

Comment:
1. Does this work if insights.sendEvents is run in its own coroutine (using coroutineScope.launch)? This is important because when a search operation starts, its job is added to the sequencer. If a new search begins, the sequencer will cancel the current search request (including the sending of insights events).
2. Another point to consider is that isLoading should be set to true before, and to false after the search() function. Additionally, sendEvents should be placed after these operations.

Reply:
1. I'm not sure how it's done on the HTTP client side. It's a `suspend` function, so it might be launch on the designated couroutine scope (the scope of HitsSearcher in our case). Do you have an idea, how we can test it? 
I tried to move it to the `GlobalScope` but we lose the testability in this case.
2. Not sure how we can achieve this, since we, once again, lose the testability. Could you assist?

Reply:
I think something like this should work:

```kotlin
   override fun searchAsync(): Job {
        return coroutineScope.launch(exceptionHandler) {
            isLoading.value = true
            val responseSearch = search()
            response.value = responseSearch
            isLoading.value = false
            sendInsightsEvent(responseSearch)
        }.also {
            sequencer.addOperation(it)
        }
    }

    private fun sendInsightsEvent(responseSearch: ResponseSearch?) {
        coroutineScope.launch {
            if (responseSearch != null && isAutoSendingHitsViewEvents) {
                val events = getViewEvents(responseSearch)
                if (events.isNotEmpty()) {
                    insights.sendEvents(events)
                }
            }
        }
    }
```

Reply:
Thanks, applied it. I keep the search and sending insights events in the same Job to ensure the testability.